crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -3084,6 +3084,9 @@ where
                 if self.enabled(Rule::YieldInInit) {
                     pylint::rules::yield_in_init(self, expr);
                 }
+                if self.enabled(Rule::YieldFromInAsyncFunction) {
+                    pylint::rules::yield_from_in_async_function(self, expr);
```

Comment:
I was confused about why the rule does not check whether `expr` is a `YieldFrom` expression. I think we can make this clearer by only accepting a `ExprYieldFrom` in `yield_from_in_async_function` and passing the `yield_from` expression here.


1. Change the match arm to `Expr::YieldFrom(yield_from) => {`

```suggestion
                    pylint::rules::yield_from_in_async_function(self, yield_from);
```